docs/operator-manual/upgrading/2.14-3.0.md

```diff
@@ -10,3 +10,75 @@ most recent minor versions (so 2.14 until 3.2 is released, and 2.13 until 3.1 is
 ## Breaking Changes
 
 So far no breaking changes have been merged. We will update this section as changes are merged.
+
+### Ignoring ApplicationSet `applyNestedSelectors` field
+
+Setting the `spec.applyNestedSelectors` field in an ApplicationSet resolves counter-intuitive behavior where filters in
+nested selectors were not applied. Starting in Argo CD 3.0, the field is ignored, and behavior is always the same as if
+`applyNestedSelectors` was set to `true`. In other words, nested selectors are always applied.
+
+#### Detection
+
+To detect if you are impacted, search your ApplicationSet controller logs for this string: `ignoring nested selector`.
+If there are no logs with this string, you are not impacted.
+
+Another way to detect if you are impacted is to run the following command:
+
+```shell
+kubectl get appsets -o=json | jq -r '.items[] | select(
+    .spec.applyNestedSelectors != true and 
+    .spec.generators[][].generators[][].generators[].selector != null
+  ) | .metadata.name'
+```
+
+The command will print the name of any ApplicationSet that has `applyNestedSelectors` unset or set to `false` and has 
+one or more nested selectors. 
+
+#### Remediation
+
+Since `applyNestedSelectors` is false by default, you can safely remove the nested selectors on ApplicationSets where
+`applyNestedSelectors` has not been explicitly set to `false`. After the selectors are removed, you can safely upgrade.
```

Comment:
If it was not set to true, that means the selectors are currently unused and can be removed. 
If I am misunderstanding, I think the sentence is confusing and should be clarified 😬 

```suggestion
Since `applyNestedSelectors` is false by default, you can safely remove the nested selectors on ApplicationSets where
`applyNestedSelectors` has not been explicitly set to `true`. After the selectors are removed, you can safely upgrade.
```